src/protobuf_impl.rs

```diff
@@ -9,45 +9,41 @@ use std::path::Path;
 use std::process::Command;
 use std::str::from_utf8;
 
-// For preference we use the protoc from bin which we bundle with the crate. If
-// there is not one suitable for the platform, then we try system protoc.
+// We use system protoc when its version matches,
+// otherwise use the protoc from bin which we bundle with the crate.
 fn get_protoc() -> String {
+    let ver_re = Regex::new(r"([0-9]+)\.([0-9]+)\.[0-9]").unwrap();
+    let output = Command::new("protoc").arg("--version").output();
+    match output {
+        Ok(o) => {
+            let caps = ver_re.captures(from_utf8(&o.stdout).unwrap()).unwrap();
+            let major = caps.get(1).unwrap().as_str().parse::<i16>().unwrap();
+            let minor = caps.get(2).unwrap().as_str().parse::<i16>().unwrap();
+            if major == 3 && minor >= 1 {
+                return "protoc".to_owned();
+            }
+            println!("The system `protoc` version mismatch, require >= 3.1.x, got {}.{}.x, fallback to the bundled `protoc`", major, minor);
```

Comment:
Should warn instead of just println? Otherwise L37 can be very confusing.

Reply:
There is no `warn` macro as in TiKV. What about changing L37 error message to "No bundled protoc for this platform"?

Reply:
But it's still confusing as developer may have `rustc` installed. Better make the version requirement more clear.

Reply:
This line already has the version printed, and L37 really means there is no bundled protoc for this platform.

Reply:
Oh, did you mean that this error message would not be printed on the console?

Reply:
What I mean is to make panic message more clear.

Reply:
How about changing the panic message in L37 to "No suitable `protoc` installed, require `protoc` >= 3.1.x. in PATH"